network/src/sync/blocks/batched.rs

```diff
@@ -42,86 +35,20 @@ struct SyncBlock {
-            if !judge_bad && node.quality.block_height as usize > our_block_height + 1 {
-                interesting_peers.push(node);
-            }
-        }
-        interesting_peers.sort_by(|x, y| y.quality.block_height.cmp(&x.quality.block_height));
-
-        // trim nodes close to us if any are > 10 blocks ahead
-        if let Some(i) = interesting_peers
-            .iter()
-            .position(|x| x.quality.block_height as usize <= our_block_height + 10)
-        {
-            interesting_peers.truncate(i + 1);
-        }
-
-        if !interesting_peers.is_empty() {
-            info!("found {} interesting peers for sync", interesting_peers.len());
-            trace!("sync interesting peers = {:?}", interesting_peers);
-        }
-
-        Ok(interesting_peers)
-    }
-
-    async fn block_locator_hashes(&mut self) -> Result<Vec<Digest>> {
-        let forks_of_interest = self.node.expect_sync().consensus.scan_forks().await?;
-        trace!("sync found {} forks", forks_of_interest.len());
-        let blocks_of_interest: Vec<Digest> = forks_of_interest.into_iter().map(|(_canon, fork)| fork).collect();
-        let mut tips_of_blocks_of_interest: Vec<Digest> = Vec::with_capacity(blocks_of_interest.len());
-        for block in blocks_of_interest {
-            if tips_of_blocks_of_interest.len() > crate::MAX_BLOCK_SYNC_COUNT as usize {
-                debug!("reached limit of blocks of interest in sync block locator hashes");
-                break;
-            }
-            let mut fork_path = self.node.storage.longest_child_path(&block).await?;
-            if fork_path.len() < 2 {
-                // a minor fork, we probably don't care
-                continue;
-            }
-            tips_of_blocks_of_interest.push(fork_path.pop().unwrap());
-        }
-        match self
-            .node
-            .storage
-            .get_block_locator_hashes(tips_of_blocks_of_interest, snarkos_consensus::OLDEST_FORK_THRESHOLD)
-            .await
-        {
-            Ok(block_locator_hashes) => Ok(block_locator_hashes),
-            Err(e) => {
-                error!("Unable to get block locator hashes from storage: {:?}", e);
-                Err(e)
-            }
-        }
-    }
-
     async fn send_sync_messages(&mut self, sync_nodes: Vec<Peer>) -> Result<usize> {
         info!("requested block information from {} peers", sync_nodes.len());
-        let block_locator_hashes = self
-            .block_locator_hashes()
-            .await?
-            .into_iter()
-            .map(|x| BlockHeaderHash(x.bytes::<32>().unwrap()))
-            .collect::<Vec<_>>();
+        let block_locator_hashes = SyncBase::block_locator_hashes(&self.base.node).await?;
         let mut future_set = vec![];
 
         for peer in sync_nodes.iter() {
-            if let Some(handle) = self.node.peer_book.get_peer_handle(peer.address) {
+            if let Some(handle) = self.base.node.peer_book.get_peer_handle(peer.address) {
```

Comment:
this is not strictly related to this PR, but: since we only need the peers' addresses, it is probably preferable to put only those in the `sync_nodes` vector, to reduce the amount of memory and atomic operations

Reply:
yeah, not worth doing since we may need more in the future, and the costs here are trivial